Cannot install on arm64 (Apple M1 Sillicon)

### OS/Arch

```bash
$ python -c 'import platform;print(platform.uname())'

uname_result(system='Darwin', node='Simions-MacBook-Pro-M1-Max.local', release='22.2.0', version='Darwin Kernel Version 22.2.0: Fri Nov 11 02:03:51 PST 2022; root:xnu-8792.61.2~4/RELEASE_ARM64_T6000', machine='arm64')
```

### Python version

```bash
$ python --version
Python 3.11.1
```

### What is the problem?

```bash
❯ pip install faust-cchardet

Collecting faust-cchardet
  Using cached faust-cchardet-2.1.13.tar.gz (331 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: faust-cchardet
  Building wheel for faust-cchardet (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for faust-cchardet (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [19 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build/lib.macosx-13.1-arm64-cpython-311
      creating build/lib.macosx-13.1-arm64-cpython-311/cchardet
      copying src/cchardet/version.py -> build/lib.macosx-13.1-arm64-cpython-311/cchardet
      copying src/cchardet/__init__.py -> build/lib.macosx-13.1-arm64-cpython-311/cchardet
      running build_ext
      building 'cchardet._cchardet' extension
      creating build/temp.macosx-13.1-arm64-cpython-311
      creating build/temp.macosx-13.1-arm64-cpython-311/src
      creating build/temp.macosx-13.1-arm64-cpython-311/src/cchardet
      clang -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include -I/opt/homebrew/opt/openssl@3/include -I/Users/simion/Library/Caches/pypoetry/virtualenvs/pydpf-SuFDIHlV-py3.11/include -I/Users/simion/.pyenv/versions/3.11.1/include/python3.11 -c src/cchardet/_cchardet.cpp -o build/temp.macosx-13.1-arm64-cpython-311/src/cchardet/_cchardet.o
      src/cchardet/_cchardet.cpp:775:10: fatal error: 'uchardet.h' file not found
      #include "uchardet.h"
               ^~~~~~~~~~~~
      1 error generated.
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for faust-cchardet
Failed to build faust-cchardet
ERROR: Could not build wheels for faust-cchardet, which is required to install pyproject.toml-based projects
```
#### Steps to reproduce the behavior
```bash 
pip install faust-cchardet
```

--- a/.github/workflows/build-and-upload-to-pypi.yml
+++ b/.github/workflows/build-and-upload-to-pypi.yml
@@ -18,7 +18,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [ubuntu-latest]
+        os: [ubuntu-latest, macos-11]
 
     steps:
       - uses: actions/checkout@v3

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -25,7 +25,7 @@ environment = {INCLUDE_PATH="/usr/local/include/uchardet", LIBRARY_PATH="/usr/lo
 before-build = [
     "git submodule sync --recursive",
     "git submodule update --init --force --recursive --depth=1",
-    "test -d {project}/src/ext/uchardet/build || (cd {project}/src/ext/uchardet/ && mkdir build && cd build && cmake -DCMAKE_MACOSX_RPATH=1 -DCMAKE_INSTALL_NAME_DIR=$LIBRARY_PATH .. && make && make install)",
+    "test -d {project}/src/ext/uchardet/build || (cd {project}/src/ext/uchardet/ && mkdir build && cd build && cmake -DCMAKE_MACOSX_RPATH=1 -DCMAKE_INSTALL_NAME_DIR=$LIBRARY_PATH -DCMAKE_BUILD_RPATH=$LIBRARY_PATH .. && make && make install)",
 ]
 
 [tool.cibuildwheel.windows]

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -13,12 +13,21 @@ test-command = [
     'python -m pytest {project}'
 ]
 
+environment = {INCLUDE_PATH="/usr/local/include/uchardet", LIBRARY_PATH="/usr/local/lib64/"}
 before-build = [
     "git submodule sync --recursive",
     "git submodule update --init --force --recursive --depth=1",
     "test -d {project}/src/ext/uchardet/build || (cd {project}/src/ext/uchardet/ && mkdir build && cd build && cmake .. && make && make install)",
 ]
-environment = {INCLUDE_PATH="/usr/local/include/uchardet", LIBRARY_PATH="/usr/local/lib64/"}
+
+[tool.cibuildwheel.macos]
+environment = {INCLUDE_PATH="/usr/local/include/uchardet", LIBRARY_PATH="/usr/local/lib/"}
+before-build = [
+    "git submodule sync --recursive",
+    "git submodule update --init --force --recursive --depth=1",
+    "test -d {project}/src/ext/uchardet/build || (cd {project}/src/ext/uchardet/ && mkdir build && cd build && cmake -DCMAKE_MACOSX_RPATH=1 .. && make && make install)",
+    "",
+]
 
 [tool.cibuildwheel.windows]
 before-build = [

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -26,7 +26,6 @@ before-build = [
     "git submodule sync --recursive",
     "git submodule update --init --force --recursive --depth=1",
     "test -d {project}/src/ext/uchardet/build || (cd {project}/src/ext/uchardet/ && mkdir build && cd build && cmake -DCMAKE_MACOSX_RPATH=1 .. && make && make install)",
-    "",
 ]
 
 [tool.cibuildwheel.windows]

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -25,7 +25,7 @@ environment = {INCLUDE_PATH="/usr/local/include/uchardet", LIBRARY_PATH="/usr/lo
 before-build = [
     "git submodule sync --recursive",
     "git submodule update --init --force --recursive --depth=1",
-    "test -d {project}/src/ext/uchardet/build || (cd {project}/src/ext/uchardet/ && mkdir build && cd build && cmake -DCMAKE_MACOSX_RPATH=1 .. && make && make install)",
+    "test -d {project}/src/ext/uchardet/build || (cd {project}/src/ext/uchardet/ && mkdir build && cd build && cmake -DCMAKE_MACOSX_RPATH=1 -DCMAKE_INSTALL_NAME_DIR=$LIBRARY_PATH .. && make && make install)",
 ]
 
 [tool.cibuildwheel.windows]

